src/python/grapl-common/grapl_common/envelope.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import uuid
+from uuid import uuid4
+
+from graplinc.grapl.api.services.v1beta1.types_pb2 import Envelope as _Envelope
+from graplinc.grapl.api.services.v1beta1.types_pb2 import Meta, ProtoUuidV4
+
+
+def proto_uuid_to_pyuuid(proto_uuid: ProtoUuidV4) -> uuid.UUID:
+    lsb = proto_uuid.lsb
+    msb = proto_uuid.msb
+    lsb_bytes = int.to_bytes(lsb, byteorder="little", length=8, signed=False)
+    msb_bytes = int.to_bytes(msb, byteorder="little", length=8, signed=False)
+
+    return uuid.UUID(bytes=lsb_bytes + msb_bytes)
+
+
+def pyuuid_to_proto_uuid(pyuuid: uuid.UUID) -> ProtoUuidV4:
+    lsb_bytes = pyuuid.bytes[:8]
+    msb_bytes = pyuuid.bytes[8:]
+    lsb = int.from_bytes(lsb_bytes, byteorder="little", signed=False)
+    msb = int.from_bytes(msb_bytes, byteorder="little", signed=False)
+
+    proto_uuid = ProtoUuidV4()
+    proto_uuid.lsb = lsb
+    proto_uuid.msb = msb
+    return proto_uuid
+
+
+def init_metadata(client_id: str) -> Meta:
```

Comment:
If we had a type to represent the `Meta` class, this could just be that class' constructor.